### ISSUE TITLE
All sensors in "Unknown" State after 2024.08 upgrade

### ISSUE THREAD

**Issue by uspino2** [wills106/homeassistant-solax-modbus#1022]
### Describe the bug

All Sensors give "Unknown" state after 2024.8.x upgrade. Everything works fine with 2024.7 and below. 

### Integration Version

2024.08.13

### Homeassistant core version

2024.08.03

### Inverter brand

Growatt

### Plugin used

plugin_growatt.py

### Serial prefix

QQK200

### Connection Method

WaveShare Ethernet

### Detailed Error Log

```
2024-08-30 13:17:33.745 INFO (MainThread) [custom_components.solax_modbus] Inverter is not connected, trying to connect
2024-08-30 13:17:33.745 DEBUG (MainThread) [custom_components.solax_modbus] Trying to connect to Inverter at 10.10.10.119:502
2024-08-30 13:17:33.748 INFO (MainThread) [custom_components.solax_modbus] Inverter connected at 10.10.10.119:502
2024-08-30 13:17:33.748 INFO (MainThread) [custom_components.solax_modbus.plugin_growatt] growatt_modbus: trying to determine inverter type
2024-08-30 13:17:33.788 INFO (MainThread) [custom_components.solax_modbus.plugin_growatt] Read growatt_modbus 0x9 firmware number before potential swap: AL1.0ZABA
2024-08-30 13:17:33.788 INFO (MainThread) [custom_components.solax_modbus.button] hub.wakeuButton: None
2024-08-30 13:17:33.798 INFO (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus sensor groups: 1
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus group: 15
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x9 firmware_version to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc firmware_control_version to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xf language to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x16 select_baud_rate to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x17 serialnumber to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x2d rtc to block with start 0x9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] Starting new block at 0x7a 
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x7a limit_grid_export to block with start 0x7a
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0x7b grid_export_limit to block with start 0x7a
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] Starting new block at 0xbe8 
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe8 Charger_upper_soc to block with start 0xbe8
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe9 charger_switch to block with start 0xbe8
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbfe battery_type to block with start 0xbe8
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc16 battery_discharge_max_power to block with start 0xbe8
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc18 battery_charge_max_power to block with start 0xbe8
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbb9 total_pv_power to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbbb pv_voltage_1 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbbc pv_current_1 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbbd pv_power_1 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbbf pv_voltage_2 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbc0 pv_current_2 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbc1 pv_power_2 to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbcf grid_power to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbd1 grid_frequency to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbd2 inverter_voltage to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbd3 grid_current to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbd4 grid_power_va to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe1 total_forward_power to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe3 total_reverse_power to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe5 total_load_power to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbe9 today_s_power_generation to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbeb total_power_generation to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbed total_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbef today_s_pv1_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbf1 total_pv1_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbf3 today_s_pv2_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.798 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbf5 total_pv2_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbfb today_s_grid_import to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbfd total_grid_import to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xbff today_s_grid_export to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc01 total_grid_export to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc03 today_s_yield to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc05 total_yield to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc0b today_s_solar_energy to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc15 inverter_temperature to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc16 ipm_inverter_temperature to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc17 boost_temperature to block with start 0xbb9
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] Starting new block at 0xc35 
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc35 today_s_battery_output_energy to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc37 total_battery_output_energy to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc39 today_s_battery_input_energy to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc3b total_battery_input_energy to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc48 work_mode_priority to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc61 battery_voltage to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc62 battery_current to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc63 battery_soc to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc6a battery_discharge_power to block with start 0xc35
2024-08-30 13:17:33.799 DEBUG (MainThread) [custom_components.solax_modbus.sensor] adding register 0xc6c battery_charge_power to block with start 0xc35
2024-08-30 13:17:33.799 INFO (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus returning holding block: 0x9 0x33 [9, 12, 15, 22, 23, 45]
2024-08-30 13:17:33.799 INFO (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus returning holding block: 0x7a 0xbb9 [122, 123, 3000]
2024-08-30 13:17:33.799 INFO (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus returning holding block: 0xbe8 0xc1a [3048, 3049, 3070, 3094, 3096]
2024-08-30 13:17:33.799 INFO (MainThread) [custom_components.solax_modbus.sensor] growatt_modbus returning input block: 0xbb9 0xc18 [3001, 3003, 3004, 3005, 3007, 3008, 3009, 3023, 3025, 3026, 3027, 3028, 3041, 3043, 3045, 3049, 3051, 3053, 3055, 3057, 3059, 3061, 3067, 3069, 3071, 3073, 3075, 3077, 3083, 3093, 3094, 3095]
```

**Comment by TonyMathiesen** [comment]
Hi. 
This is related to issue #791. Version 2024.08.12 is working. 
You can also comment out register 3000 in the plugin_growatt.py file to get it working until fixed. 
![image](https://github.com/user-attachments/assets/99981241-e1b8-4979-92a4-f68b98a3d9f3)

**Comment by wills106** [comment]
I have released [2024.08.14](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.08.14) with the change, I'll leave the issue open until confirmed it works.

**Comment by uspino2** [comment]
All working now. Thank you @wills106 and @TonyMathiesen.

**Comment by uspino2** [comment]
One change I notice, no Battery Working mode in Controls. With 2024.07.X I had:

![Screenshot 2024-08-31 at 5 34 10 PM](https://github.com/user-attachments/assets/cb6af9c7-df45-49c5-a2ef-db657278bb25)

With 2024.08.14 only "Sync RTC" appears... Is that intentional? (What is "Sync RTC", by the way?)

**Comment by TonyMathiesen** [comment]
"Battery First Maximum SOC" & "Charger Upper SOC" er replaced with the EMS / Battery controls under configuration
![image](https://github.com/user-attachments/assets/887f213d-b99b-4123-b6ac-d50f2655c3ff)

In order to align the naming with web UI and APP:
![image](https://github.com/user-attachments/assets/beeee450-17ef-45f8-9713-047e88e13770)

**Comment by uspino2** [comment]
This is what shows up with my Inverter / Battery. There's no "Discharging" controls...

![Screenshot 2024-08-31 at 6 45 02 PM](https://github.com/user-attachments/assets/d505c798-7b7a-4156-a5cb-6bac8be8be3f)

**Comment by TonyMathiesen** [comment]
Found it. If you add GEN4 on line 233 and 249 in the plugin_growatt.py file. Then they will be shown. My inverter uses the GEN3 for the filtering, and I forgot to add the GEN4 on the configuration inputs, there are under the sensors. 

![image](https://github.com/user-attachments/assets/16f73ebf-67b3-4c47-b700-d0d1bd5bdf93)

**Comment by uspino2** [comment]
Perfect, thank you!

**Comment by wills106** [comment]
I have released [2024.09.1](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.09.1) with this change.

**Comment by wills106** [comment]
Closing as corrected.
